tiling.py

```diff
@@ -398,14 +443,21 @@ def fundamental_domains(self):
     @property
     def translational_fundamental_domains(self):
         for i in range(self._diameter):
-            pos = np.array((i - self._radius)) + np.array(self._center)
-            translate = utils.translation_matrix(
-                abs(pos) * self._group.translational_subgroup_basis[1 if pos > 0 else 0])
+            for j in range(self._vdiameter):
+                pos = np.array((i - self._radius, (j - self._vradius) if self._vdiameter != 1 else 0)) + np.array(self._center)
+                #skip the tfds under the floor
+                if self._group.flag_ceiling and not self._group.flag_floor and pos[1] < 0:
+                    continue;
+                # skip the tfds above the floor
+                if self._group.flag_floor and not self._group.flag_ceiling and pos[1] > 0:
+                    continue;
+                translate = utils.translation_matrix(
+                    abs(pos[0]) * self._group.translational_subgroup_basis[1 if pos[0] > 0 else 0] + pos[1] * self._group.translational_subgroup_basis[2])
 
-            prism = shapes.Prism(self._group.height, *self._group.translational_fd_vertices)
+                prism = shapes.Prism(self._group.height*2, *self._group.translational_fd_vertices)
 
-            prism.transform(translate)
-            yield pos, translate, prism
+                prism.transform(translate)
+                yield pos, translate, prism
```

Comment:
```
vdiameter = diameter if floor and ceiling else 1
```

then your inner loop becomes:

```
for j in range vdiameter:
    pos = np.array((i - self.radius, j - self.radius)) + np.array((self._center))
    ...
```
you'll need to rework some of the logic here, but I think it's pretty straightforward and should be really simple.